R/get_national_data.R

```diff
@@ -7,35 +7,37 @@
 #'  in a standard format.
 #'
 #' @param country A character string specifying the country to get data from.
-#'  Not case or language dependent. Defaults to all countries.
+#' Not case dependent. Name should be the English name. For a list of
+#' options see `available_datasets`.
 #' @param source A character string specifying the data source: "WHO", or
 #'  "ECDC". Not case dependent. Defaults to WHO.
+#' @param steps Logical, defaults to FALSE. Should all processing and cleaning
+#' steps be kept and output in a list.
+#' @param verbose Logical, defaults to `TRUE`. Should verbose processing
+#' messages and warnings be returned.
+#' @param ... additional arguments to pass to Country classes.
 #' @return A tibble with data related to cases, deaths, hospitalisations,
 #'  recoveries and testing.
-#' @inheritParams get_regional_data
 #' @importFrom dplyr group_by arrange select ungroup do everything
 #' @importFrom tidyr drop_na fill
 #' @importFrom countrycode countryname
 #' @export
-#' @examples
-#' \dontrun{
-#' # set up a data cache
-#' start_using_memoise()
-#'
-#' get_national_data(country = "canada", source = "ecdc", steps = TRUE)
-#' }
 #'
 get_national_data <- function(country, source = "who", steps = FALSE,
-                              verbose = TRUE) {
+                              verbose = TRUE, ...) {
 
-  # check data availability and define list
-  source <- new_covidregionaldata(source, level = "1", verbose = verbose)
+  # check data availability and initiate country class if avaliable
+  nation_class <- check_country_avaliable(
+    country = source, level = "1",
+    totals = FALSE, localise = TRUE,
+    verbose = verbose, steps = steps, ...
+  )
 
   # download and cache raw data
-  source <- download_regional(source, verbose = verbose)
+  nation_class$download_data()
```

Comment:
Add back in progress messages from S3 version with verbose switch

Reply:
verbose is an attribute of the country class and set at initialisation so it should work in the same was as the s3 method